Cabal/src/Distribution/Simple/Program/Ar.hs

```diff
@@ -68,17 +68,25 @@ createArLibArchive verbosity lbi targetPath files = do
   --     do that. We have duplicates because of modules like "A.M" and "B.M"
   --     both make an object file "M.o" and ar does not consider the directory.
   --
+  --  -- llvm-ar, which GHC >=9.4 uses on Windows, supports a "L" modifier
+  --     in "q" mode which compels the archiver to add the members of an input
+  --     archive to the output, rather than the archive itself. This is
+  --     necessary as GHC may produce .o files that are actually archives. See
+  --     https://gitlab.haskell.org/ghc/ghc/-/issues/21068.
+  --
   -- Our solution is to use "ar r" in the simple case when one call is enough.
   -- When we need to call ar multiple times we use "ar q" and for the last
   -- call on OSX we use "ar qs" so that it'll make the index.
 
   let simpleArgs  = case hostOS of
              OSX -> ["-r", "-s"]
+             _ | dashLSupported -> ["-qL"]
```

Comment:
is this (and will be) harmless on linux?

Reply:
Indeed. I have seen no traditional use of `.a` archives which would want the non-`-L` behavior; you always want the merging behavior that `-L` introduces. It's somewhat incredible that binutils' `ar` does not support it yet.